### PR TITLE
[RFR] Fix AutocompleteArrayInput options

### LINKED ISSUE #3477: AutocompleteArrayInput not accepting props
**What you were expecting:**
On passing prop `fullWidth` in `options` of `AutocompleteArrayInput` component, ChipInput to have full wdth. I guess no props are accessible by `AutocompleteArrayInput`.

**What happened instead:**
passing `fullWidth` didn't work.

**Related code:**
```
<AutoCompleteArrayInput options={{
    fullWidth: true
}} />
```
**Steps to reproduce**
1- Open up any post
2- Go to `MISCELLANEOUS`
3- ChipInput doesn't has full width

**CodeSandbox**
https://codesandbox.io/embed/simple-jkp47

**Additional Information**
Even after sending `options` prop, `AutocompleteArrayInput` got `options: { }` (empty object) when checked using `console log`.

**Environment**

* React-admin version:
* Last version that did not exhibit the issue (if applicable):
* React version:
* Browser:
* Stack trace (in case of a JS error):

### PR DESCRIPTION
**Pull request by djhi**
Fixes #3477
image

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/AutocompleteArrayInput.js [hunk 1]
```diff
@@ -247,6 +250,11 @@ export class AutocompleteArrayInput extends React.Component {
             ref(input);
         };
 
+        const finalOptions = {
+            fullWidth,
+            ...options,
```
Comment (Kmaschta) [inline packages/ra-ui-materialui/src/input/AutocompleteArrayInput.js]: So we ignore all others options?

File: docs/Inputs.md [hunk 2]
```diff
@@ -297,11 +297,12 @@ Lastly, `<AutocompleteArrayInput>` renders a [material-ui-chip-input](https://gi
 {% raw %}
 ```jsx
 <AutocompleteArrayInput source="category" options={{
-    fullWidth: true,
+    fullWidthInput: true,
 }} />
 ```
 {% endraw %}
 
+**Tip**: Like many other inputs, `<AutocompleteArrayInput>` accept a `fullWidth` prop.
```
[the remark this answers is not in the document]
Comment (djhi) [reply docs/Inputs.md]: No, it's `fullWidth`

File: docs/Inputs.md [hunk 3]
```diff
@@ -341,6 +342,7 @@ If you need to override the props of the suggestions container (a `Popper` eleme
 | `setFilter` | Optional | `Function` | null | A callback to inform the `searchText` has changed and new `choices` can be retrieved based on this `searchText`. Signature `searchText => void`. This function is automatically setup when using `ReferenceInput`.  |
 | `suggestionComponent` | Optional | Function | `({ suggestion, query, isHighlighted, props }) => <div {...props} />` | Allows to override how the item is rendered.  |
 | `shouldRenderSuggestions` | Optional | Function | `() => true` | A function that returns a `boolean` to determine whether or not suggestions are rendered. Use this when working with large collections of data to improve performance and user experience. This function is passed into the underlying react-autosuggest component. Ex.`(value) => value.trim() > 2` |
+| `fullWith` | Optional | Boolean | If `true`, the input will take all the form width
```
[the remark this answers is not in the document]
Comment (djhi) [reply docs/Inputs.md]: Same

File: packages/ra-ui-materialui/src/input/AutocompleteArrayInput.js (hunk 1 above)
Comment (djhi) [reply packages/ra-ui-materialui/src/input/AutocompleteArrayInput.js]: Have you missed the `...options` ?

Comment (Kmaschta) [reply packages/ra-ui-materialui/src/input/AutocompleteArrayInput.js]: Nvm I thought it was destructuring, but it isn't !

File: docs/Inputs.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kmaschta) [reply docs/Inputs.md]: Ok my bad. I would just add a precision: "fullWidth is related to the field and fillWidthInput is related to the ChipInput" to be extra clear about the meaning of these props.

[the remark this answers is not in the document]
Comment (djhi) [reply docs/Inputs.md]: Nope, we won't repeat the `material-ui-chip-input` documentation

**Pull request merged**